staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
```

Comment:
@liggitt I think I got what you mean, and as we already have tokenSource provided by the auth provider, we need remove the `WrapTransport` from auth provider interface,  make the `WrapTransport` immutable to the auth provider, and generic to all the auth type when wrap the `http.RoundTripper`.

And also, add a comment, if BearerTokenSource and provider config coexists, we need prefer the BearerTokenSource you provided.
How do you think about this?